contracts/apps/voting/VotingApp.sol

```diff
@@ -0,0 +1,190 @@
+pragma solidity 0.4.15;
+
+import "./EVMScriptRunner.sol";
+import "../App.sol";
+
+import "../../common/Initializable.sol";
+import "../../common/MiniMeToken.sol";
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract VotingApp is App, Initializable, EVMScriptRunner {
+    using SafeMath for uint256;
+
+    MiniMeToken public token;
+    uint256 public supportRequiredPct;
+    uint256 public minAcceptQuorumPct;
+    uint64 public voteTime;
+
+    uint256 constant PCT_BASE = 10 ** 18;
+
+    enum VoterState { Absent, Yea, Nay }
+
+    struct Voting {
+        address creator;
+        uint64 startDate;
+        uint256 snapshotBlock;
+        uint256 yea;
+        uint256 nay;
+        uint256 totalVoters;
+        bytes executionScript;
+        bool open;
+        bool executed;
+        mapping (address => VoterState) voters;
+    }
+
+    Voting[] votings;
+
+    event StartVote(uint256 indexed votingId);
+    event CastVote(uint256 indexed votingId, address indexed voter, bool supports);
+    event ExecuteVote(uint256 indexed votingId);
+
+    /**
+    * @notice Initializes VotingApp (some parameters won't be modifiable after being set)
+    * @param _token MiniMeToken address that will be used as governance token
+    * @param _supportRequiredPct Percentage of voters that must support a voting for it to succeed (expressed as a 10^18 percetage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _minAcceptQuorumPct Percetage of total voting power that must support a voting  for it to succeed (expressed as a 10^18 percetage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _voteTime Seconds that a voting will be open for token holders to vote (unless it is impossible for the fate of the vote to change)
+    */
+    function initialize(MiniMeToken _token, uint256 _supportRequiredPct, uint256 _minAcceptQuorumPct, uint64 _voteTime) onlyInit {
+        initialized();
+
+        require(_supportRequiredPct <= PCT_BASE);
+        require(_supportRequiredPct >= _minAcceptQuorumPct);
+
+        token = _token;
+        supportRequiredPct = _supportRequiredPct;
+        minAcceptQuorumPct = _minAcceptQuorumPct;
+        voteTime = _voteTime;
+
+        votings.length += 1;
+    }
+
+    /**
+    * @notice Change minimum acceptance quorum to `_minAcceptQuorumPct`
+    * @param _minAcceptQuorumPct New acceptance quorum
+    */
+    function changeMinAcceptQuorumPct(uint256 _minAcceptQuorumPct) auth external {
+        require(supportRequiredPct >= _minAcceptQuorumPct);
+        minAcceptQuorumPct = _minAcceptQuorumPct;
+    }
+
+    /**
+    * @notice Create new voting to execute `_executionScript`
+    * @param _executionScript EVM script to be executed on approval
+    * @return votingId id for newly created vote
+    */
+    function newVoting(bytes _executionScript) auth external returns (uint256 votingId) {
+        votingId = votings.length++;
```

Comment:
Nice, I'll start using this method of determining the id

Reply:
👍. In this case it is probably fine, but we need to be very careful the client handles chain reorgs well if they happen. We don't want users voting on an incorrect votingId.